bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -163,61 +157,76 @@ public BigtableDataGrpcClient(
+    return performAsyncRpc(request, BigtableServiceGrpc.METHOD_MUTATE_ROWS, ARE_RETRYABLE_MUTATIONS);
   }
 
   @Override
   public CheckAndMutateRowResponse checkAndMutateRow(CheckAndMutateRowRequest request)
       throws ServiceException {
-    return clientCallService.blockingUnaryCall(createCheckAndMutateRowCall(request), request);
+    return performBlockingRpc(
+        request, BigtableServiceGrpc.METHOD_CHECK_AND_MUTATE_ROW, IS_RETRYABLE_CHECK_AND_MUTATE);
   }
 
   @Override
   public ListenableFuture<CheckAndMutateRowResponse> checkAndMutateRowAsync(
       CheckAndMutateRowRequest request) {
     expandPoolIfNecessary(this.bigtableOptions.getChannelCount());
-    return clientCallService.listenableAsyncCall(createCheckAndMutateRowCall(request), request);
+    return performAsyncRpc(
+        request, BigtableServiceGrpc.METHOD_CHECK_AND_MUTATE_ROW, IS_RETRYABLE_CHECK_AND_MUTATE);
   }
 
-  private ClientCall<MutateRowRequest, Empty> createMutateRowCall(MutateRowRequest request) {
-    return createRetryableCall(BigtableServiceGrpc.METHOD_MUTATE_ROW,
-      IS_RETRYABLE_MUTATION, request);
-  }
+  private <ReqT, RespT> RespT performBlockingRpc(
+      ReqT request,
+      MethodDescriptor<ReqT, RespT> method,
+      Predicate<ReqT> retryablePredicate) {
+    CancellationToken token = new CancellationToken();
+    RetryableRpc<ReqT, RespT> mutateRowAsync =
+        BigtableAsyncUtilities.createAsyncUnaryCall(channelPool, clientCallService, method, token);
 
-  private ClientCall<MutateRowsRequest, MutateRowsResponse>
-      createMutateRowsCall(MutateRowsRequest request) {
-    return createRetryableCall(BigtableServiceGrpc.METHOD_MUTATE_ROWS, ARE_RETRYABLE_MUTATIONS,
-      request);
+    try {
+      return BigtableAsyncUtilities.getUnchecked(
+          performRetryableRpc(request, mutateRowAsync, retryablePredicate));
+    } catch (Throwable t) {
+      token.cancel();
+      throw Throwables.propagate(t);
+    }
   }
 
-  private ClientCall<CheckAndMutateRowRequest, CheckAndMutateRowResponse>
-      createCheckAndMutateRowCall(CheckAndMutateRowRequest request) {
-    return createRetryableCall(BigtableServiceGrpc.METHOD_CHECK_AND_MUTATE_ROW,
-      IS_RETRYABLE_CHECK_AND_MUTATE, request);
-  }
+  private <ReqT, RespT> ListenableFuture<RespT> performAsyncRpc(
+      ReqT request,
+      MethodDescriptor<ReqT, RespT> method,
+      Predicate<ReqT> predicate) {
+    CancellationToken token = new CancellationToken();
+    RetryableRpc<ReqT, RespT> mutateRowAsync =
+        BigtableAsyncUtilities.createAsyncUnaryCall(
+            channelPool, clientCallService, method, token);
 
-  private <ReqT, RespT> ClientCall<ReqT, RespT> createRetryableCall(
-      MethodDescriptor<ReqT, RespT> method, Predicate<ReqT> isRetryable, ReqT request) {
+    return performRetryableRpc(request, mutateRowAsync, predicate);
+  }
+  
+  private <ReqT, RespT> ListenableFuture<RespT> performRetryableRpc(
```

Comment:
Not sure what would be better, but performRetryableRpc kind of clashes with (performBlockingRpc | performAsyncRpc) and BigtableAsyncUtilities.performAsyncRpc